### PR TITLE
fix Kafka::TransactionManager#send_offsets_to_txn

### PR DESCRIPTION
**Pull request by stasCF**
- Fix for handling `TxnOffsetCommitResponse` according to Kafka protocol: https://kafka.apache.org/protocol#The_Messages_TxnOffsetCommit
- Fix for txn_offset_commit: it should be called from group coordinator according to [native Java client](https://github.com/apache/kafka/blob/b937ec75677f8af13bf6fda686f07e9c62cdd20f/clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java#L1705), otherwise we got an error.

### CODE REVIEW

**Comment by stasCF** [comment]
Hey @dasch, can you give a little advice? I have a failing functional spec for 3 Kafka versions in CircleCI:
```
wurstmeister/kafka:2.11-0.11.0.3
wurstmeister/kafka:2.11-2.0.1
wurstmeister/kafka:2.12-2.4.0
```
The problem is I cannot reproduce this fail locally for any of these versions and it appears my code is not related with the failing spec. I run my specs in the same way as in CircleCI step.
What's the next steps should be?

**Comment by dasch** [comment]
@stasCF sorry about that – the functional specs are unfortunately a bit flaky. I've restarted them.

If someone was inclined to make them more robust I wouldn't at all mind ;)

File: lib/kafka/protocol/txn_offset_commit_response.rb [hunk 1]
```diff
@@ -1,17 +1,46 @@
...
 
-      def initialize(error_code:)
-        @error_code = error_code
+      def initialize(errors:)
+        @errors = errors
       end
 
       def self.decode(decoder)
         _throttle_time_ms = decoder.int32
-        error_code = decoder.int16
```
Comment (dasch) [inline lib/kafka/protocol/txn_offset_commit_response.rb:13]: I don't get how this worked in the first place...

**Review by dasch — APPROVED** [review]
Code looks good. Have you verified that this branch works in your staging or production setup?

**Comment by stasCF** [comment]
@dasch yes, we run our fork of ruby-kafka in production with this fix included for a couple of weeks and I can confirm it works as expected.

**Comment by dasch** [comment]
Awesome, thanks!

**Comment by dasch** [comment]
If you add a line to the changelog I'll go ahead and merge this.

**Comment by stasCF** [comment]
@dasch done. thanks for your help!

**Pull request merged**